src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -5566,6 +5568,247 @@ void GetVRFInputData(EnergyPlusData &state, bool &ErrorsFound)
+                    }
+                    ErrorsFound = true;
+                }
+            } else if (HeatingCoilPresent) {
+                if (VRFTUOAMixerMixedNodeNum != heatCoilAirInNode) {
+                    ShowSevereError(state, cTerminalUnitType + " \"" + cTUName + "\"");
+                    ShowContinueError(state,
+                                      "... For draw thru or no fan when an OA mixer is specified and a cooling coil is not present "
+                                      "the OA mixer mixed air node name must match the heating coil inlet node name.");
+                    if (VRFTUOAMixerMixedNodeNum > 0 && heatCoilAirInNode > 0) {
+                        ShowContinueError(state, "... OA mixer mixed air node name = " + nodeID(VRFTUOAMixerMixedNodeNum) + ".");
+                        ShowContinueError(state, "... Heating coil inlet node name = " + nodeID(heatCoilAirInNode) + ".");
+                    }
+                    ErrorsFound = true;
+                }
+            }
+        } else { // OAMixer not used
+            if (CoolingCoilPresent) {
+                if (VRFTUInletNodeNum != coolCoilAirInNode) {
+                    ShowSevereError(state, cTerminalUnitType + " \"" + cTUName + "\"");
+                    ShowContinueError(
+                        state,
+                        "... For draw thru or no fan when no OA mixer is specified and a cooling coil is present the terminal unit inlet "
+                        "node name must match the cooling coil inlet node name.");
+                    if (VRFTUInletNodeNum > 0 && coolCoilAirInNode > 0) {
+                        ShowContinueError(state, "... Terminal unit inlet node name = " + nodeID(VRFTUInletNodeNum) + ".");
+                        ShowContinueError(state, "... Cooling coil inlet node name = " + nodeID(coolCoilAirInNode) + ".");
+                    }
+                    ErrorsFound = true;
+                }
+            } else if (HeatingCoilPresent) {
+                if (VRFTUInletNodeNum != heatCoilAirInNode) {
+                    ShowSevereError(state, cTerminalUnitType + " \"" + cTUName + "\"");
+                    ShowContinueError(state,
+                                      "... For draw thru or no fan when no cooling coil or OA mixer is specified the terminal unit inlet "
+                                      "node name must match the heating coil inlet node name.");
+                    if (VRFTUInletNodeNum > 0 && heatCoilAirInNode > 0) {
+                        ShowContinueError(state, "... Terminal unit inlet node name = " + nodeID(VRFTUInletNodeNum) + ".");
+                        ShowContinueError(state, "... Heating coil inlet node name = " + nodeID(heatCoilAirInNode) + ".");
+                    }
+                    ErrorsFound = true;
+                }
+            }
+        }
+    }
+    if (FanPlace == DataHVACGlobals::BlowThru && !OAMixerUsed) {
```

Comment:
I'm not seeing a check when `FanPlace == DataHVACGlobals::BlowThru && OAMixerUsed`

Reply:
The logic here works, but it seems somewhat complicated, so many nested ifs, some with dual comparisons. Wouldn't it be easier to just start with the inlet node as the `currentNode` and walk through the components one by one, skipping those that don't exist and check that the next component's inlet matches `currentNode`, then update `currentNode` to the current component's outlet and mode to the next component, and so on.

Reply:
It's fine like it is.